client/src/js/services/journalDataviewService.js

```diff
@@ -0,0 +1,205 @@
+angular.module('bhima.services')
+.service('JournalDataviewService', ['JournalDataLoaderService', 'JournalManagerService', '$translate',
+  function (dataLoaderService, managerService, $translate) {
+    var dataviewService = this;
+    var inlineFilter = true;
+
+    dataviewService.aggregates = true;    
+    dataviewService.gimp = new Slick.Data.GroupItemMetadataProvider(); //plugin to group metadata for slick grid
+    dataviewService.dataLoaderService = dataLoaderService;
+    dataviewService.managerService = managerService;
+    dataviewService.dataview = new Slick.Data.DataView({
+      groupItemMetadataProvider: dataviewService.gimp,
+      inlineFilter: inlineFilter
+    });
+
+
+    dataviewService.setAggregate = function setAggregate (ans){
+      dataviewService.aggregates = ans;
+    };
+
+    dataviewService.getAggregate = function getAggregate (){
+      return dataviewService.aggregates;
+    };
+
+    dataviewService.subscribeToOnRowCountChanged = function subscribeToOnRowCountChanged (grid){
+      if(!grid){throw 'Undefined grid';}
+      dataviewService.dataview.onRowCountChanged.subscribe(function (e, args) {
+        grid.updateRowCount();
+        grid.render();
+      });      
+    };
+
+    dataviewService.subscribeToOnRowsChanged = function subscribeToOnRowsChanged (grid){
+      if(!grid){throw 'Undefined grid';}
+      dataviewService.dataview.onRowsChanged.subscribe(function (e, args) {
+        grid.invalidateRows(args.rows);
+        grid.render();
+      });
+    };
+
+    dataviewService.populate = function populate () {
+      return dataviewService.dataLoaderService.loadJournalRecord()
+      .then(function (data){
+        data.journalRecord.data = data.journalRecord.data.map(function (item) {
+          item.trans_date = new Date(item.trans_date);
+          return item;
+        });
+        dataviewService.dataview.beginUpdate();
+        dataviewService.dataview.setItems(data.journalRecord.data, 'uuid');
+        dataviewService.dataview.endUpdate();   
+      });      
+    };
+
+    dataviewService.addNewItem = function addNewItem(row){
+      dataviewService.dataview.addItem(row);
+    };
+
+    dataviewService.getItem = function getItem(row){
+      return dataviewService.dataview.getItem(row);
+    };
+
+    dataviewService.getItems = function getItem(){
```

Comment:
I notice that you duplicate a lot of methods from dataview --> dataviewService.  Is this easier than having a controller call `dataviewService.dataview.getItems()`? 

If you think aliasing is advantageous, I recommend you use the exact same API as dataview. So, you can write this:
```js
// alias dataview properties for faster lookups
service.getItems = service.dataview.getItems;
service.getItem = service.dataview.getItem;
service.addItem = service.dataview.addItem;
service.updateItem = service.dataview.updateItem;
// etc...
```
Avoiding changing the API will cause less confusion among programmers.  It also allows you to write compact code.  However, I'm not certain why a controller shouldn't just call `DataviewService.dataview.getItems()` anyway.